### PR TITLE
Update the release script [skip ci]

### PR DESCRIPTION
**Pull request by NvTimLiu**
Argument for the spark version of the final build and aggregation

Signed-off-by: Tim Liu <timl@nvidia.com>

### CODE REVIEW

File: jenkins/deploy.sh [hunk 1]
```diff
@@ -82,15 +83,15 @@ $DEPLOY_CMD -Durl=$SERVER_URL -DrepositoryId=$SERVER_ID \
 # Distribution jar is a shaded artifact so use the reduced dependency pom.
 $DEPLOY_CMD -Durl=$SERVER_URL -DrepositoryId=$SERVER_ID \
             $SRC_DOC_JARS \
-            -Dfile=$FPATH.jar -DgroupId=com.nvidia -DartifactId=$ART_ID -Dversion=$ART_VER -DpomFile=./dist/target/dependency-reduced-pom.xml
```
Comment (NvTimLiu) [inline jenkins/deploy.sh:85]: We removed the `depednecy-reduce` script from dist pom.xml from branch-21.12, so not such `./dist/target/dependency-reduced-pom.xml` will be generated.

File: jenkins/deploy.sh [hunk 2]
```diff
@@ -55,14 +56,14 @@ FPATH="$DIST_PL/target/$ART_ID-$ART_VER"
...
+FINAL_AGG_VERSION_TOBUILD=${FINAL_AGG_VERSION_TOBUILD:-'301'}
 ###### Choose the deploy command ######
 
 if [ "$SIGN_FILE" == true ]; then
     # No javadoc and sources jar is generated for shade artifact only. Use 'sql-plugin' instead
     SQL_ART_ID=`mvn help:evaluate -q -pl $SQL_PL -Dexpression=project.artifactId -DforceStdout`
     SQL_ART_VER=`mvn help:evaluate -q -pl $SQL_PL -Dexpression=project.version -DforceStdout`
-    JS_FPATH="${SQL_PL}/target/${SQL_ART_ID}-${SQL_ART_VER}"
+    JS_FPATH="${SQL_PL}/target/spark${FINAL_AGG_VERSION_TOBUILD}/${SQL_ART_ID}-${SQL_ART_VER}"
     SRC_DOC_JARS="-Dsources=${JS_FPATH}-sources.jar -Djavadoc=${JS_FPATH}-javadoc.jar"
```
Comment (tgravescs) [inline jenkins/deploy.sh:67]: I never realized we were doing this, it works but techinically not correct. I'll file a separate issue for that though to see if we want to handle differently.

**Review by tgravescs — APPROVED** [review]

**Comment by tgravescs** [comment]
build

File: jenkins/deploy.sh [hunk 3]
```diff
@@ -55,14 +56,14 @@ FPATH="$DIST_PL/target/$ART_ID-$ART_VER"
...
-
+FINAL_AGG_VERSION_TOBUILD=${FINAL_AGG_VERSION_TOBUILD:-'301'}
 ###### Choose the deploy command ######
 
 if [ "$SIGN_FILE" == true ]; then
     # No javadoc and sources jar is generated for shade artifact only. Use 'sql-plugin' instead
     SQL_ART_ID=`mvn help:evaluate -q -pl $SQL_PL -Dexpression=project.artifactId -DforceStdout`
     SQL_ART_VER=`mvn help:evaluate -q -pl $SQL_PL -Dexpression=project.version -DforceStdout`
-    JS_FPATH="${SQL_PL}/target/${SQL_ART_ID}-${SQL_ART_VER}"
+    JS_FPATH="${SQL_PL}/target/spark${FINAL_AGG_VERSION_TOBUILD}/${SQL_ART_ID}-${SQL_ART_VER}"
```
Comment (NvTimLiu) [inline jenkins/deploy.sh:66]: we append spark version to the target path since branch-21.12.

File: jenkins/deploy.sh (hunk 2 above)
Comment (NvTimLiu) [reply jenkins/deploy.sh:67]: > 

Thanks!

**Pull request merged**